mee-components/src/layouts/Layout.astro

```diff
@@ -141,4 +161,97 @@ import 'bootstrap-icons/font/bootstrap-icons.css';
+  }
 </style>
+
+<script>
+  const scrollWrap = document.querySelector('.layout');
+  const scrollVerticalIndicator = document.querySelector('.layoutVerticalScroll') as HTMLElement;
+  const scrollHorizontalIndicator = document.querySelector('.layoutHorizontalScroll') as HTMLElement;
+
+  const setHeight = () => {
+    if (scrollWrap && scrollVerticalIndicator) {
+      const containerHeight = document.documentElement.clientHeight;
+      const contentHeight = scrollWrap.scrollHeight;
+      const scrollTop = scrollWrap.scrollTop;
+
+      if (contentHeight <= containerHeight) {
+        scrollVerticalIndicator.style.display = 'none';
+      } else {
+        scrollVerticalIndicator.style.display = 'block';
+        scrollVerticalIndicator.style.height = `${(containerHeight / contentHeight) * 100}%`;
+        scrollVerticalIndicator.style.top = `${(scrollTop / contentHeight) * 100}%`;
+      }
+    }
+  };
+
+  const setWidth = () => {
+    if (scrollWrap && scrollHorizontalIndicator) {
+      const containerWidth = document.documentElement.clientWidth;
+      const contentWidth = scrollWrap.scrollWidth;
+      const scrollLeft = scrollWrap.scrollLeft;
+
+      if (contentWidth <= containerWidth) {
+      } else {
+        scrollHorizontalIndicator.style.width = `${(containerWidth / contentWidth) * 100}%`;
+        scrollHorizontalIndicator.style.left = `${(scrollLeft / contentWidth) * 100}%`;
+      }
+    }
+  };
+
+   document.addEventListener('DOMContentLoaded', setHeight);
+   document.addEventListener('DOMContentLoaded', setWidth);
+
+   window.addEventListener('scroll', () => {
+    let timer;
+
+    clearTimeout(timer);
+
+    const scrollVerticalIndicator = document.querySelector('.layoutVerticalScroll') as HTMLElement;
+    const scrollHorizontalIndicator = document.querySelector('.layoutHorizontalScroll') as HTMLElement;
+
+    if (scrollVerticalIndicator && scrollWrap) {
+      const containerHeight = document.documentElement.clientHeight;
+
+      scrollVerticalIndicator.style.top = 
+      `${(window.scrollY / (document.body.scrollHeight - window.innerHeight)) * (100 - containerHeight / scrollWrap.scrollHeight * 100)}%`;
+
+      scrollVerticalIndicator.classList.remove('opacity-0');
+      scrollVerticalIndicator.classList.add('opacity-35');
+
+      timer = setTimeout(() => {
+        scrollVerticalIndicator.classList.remove('opacity-35');
+        scrollVerticalIndicator.classList.add('opacity-0');
+      }, 2000);
+    }
+
+    if (scrollHorizontalIndicator && scrollWrap) {
+      const containerWidth = document.documentElement.clientWidth;
+
+      scrollHorizontalIndicator.style.left = `${(document.body.scrollLeft - window.innerWidth ) * (100 - containerWidth / scrollWrap.scrollWidth * 100)}%`;
+
+      scrollVerticalIndicator.classList.remove('opacity-0');
+      scrollVerticalIndicator.classList.add('opacity-35');
+
+      timer = setTimeout(() => {
+        scrollVerticalIndicator.classList.remove('opacity-35');
```

Comment:
Do we need opacity-35 here because it is already in the --color-scrollbar variable